### PR TITLE
[17.0][IMP] RMA - refactor to use procurement run

### PR DESCRIPTION
**Pull request by victoralmau**
FWP from 16.0: https://github.com/OCA/rma/pull/402 + https://github.com/OCA/rma/pull/392

Extra changes:
- Change reception_move_ids to reception_move_id
- Add test_rma_replace_pick_ship
- Code and method reduction to simplify logic
- Set route_ids (in/out) from procurements

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT48789

### CODE REVIEW

File: rma/models/stock_picking.py [hunk 1]
```diff
@@ -30,8 +30,8 @@ def copy(self, default=None):
 
     def action_view_rma(self):
         self.ensure_one()
-        action = self.sudo().env.ref("rma.rma_action").read()[0]
-        rma = self.move_ids.mapped("rma_ids")
+        action = self.env["ir.actions.act_window"]._for_xml_id("rma.rma_action")
+        rma = self.move_lines.mapped("rma_ids")
```
Comment (chienandalu) [inline rma/models/stock_picking.py]:
```suggestion
        rma = self.move_ids.mapped("rma_ids")
```

**Review by chienandalu — CHANGES_REQUESTED** [review]

File: rma/models/stock_picking.py (hunk 1 above)
Comment (pedrobaeza) [reply rma/models/stock_picking.py]: In fact:

```suggestion
        rma = self.move_ids.rma_ids
```

**Review by chienandalu — APPROVED** [review]
:+1:

**Comment by pedrobaeza** [comment]
Testing in the runboat http://oca-rma-17-0-pr397-605691345d56.runboat.odoo-community.org to return a delivery order marking the RMA option, I get:

imagen

**Comment by victoralmau** [comment]
Reviewing the error you indicate, I see that it is a side effect of something that already happens currently in v17 and v16, if the user does NOT have the locations permission, the picking that is created does NOT have the correct data (location and picking_type_id).

This does not happen in tests because the user that is used does have this permission, I will review it to fix it.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge major

**Pull request merged**